src/components/CampaignInitialisation/CampaignInitialisationScreen.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useCallback,
+  useState
+} from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { StyleSheet, View } from "react-native";
+import { size } from "../../common/styles";
+import { NavigationProps } from "../../types";
+import { UpdateByRestartingContent } from "./UpdateByRestartingContent";
+import { UpdateFromAppStoreContent } from "./UpdateFromAppStoreContent";
+import { LoadingView } from "../Loading";
+import { useUpdateCampaignConfig } from "../../hooks/useUpdateCampaignConfig/useUpdateCampaignConfig";
+import { useAuthenticationContext } from "../../context/auth";
+import { useCheckVersion } from "../../hooks/useCheckVersion/useCheckVersion";
+import { useCheckUpdates } from "../../hooks/useCheckUpdates";
+
+const styles = StyleSheet.create({
+  wrapper: {
+    ...StyleSheet.absoluteFillObject,
+    justifyContent: "center",
+    alignItems: "center",
+    paddingHorizontal: size(5)
+  },
+  content: {
+    width: 512,
+    maxWidth: "100%",
+    marginTop: -size(4)
+  }
+});
+
+const RETRY_UPDATE_TIMES = 3;
+
+export const CampaignInitialisationScreen: FunctionComponent<NavigationProps> = ({
+  navigation
+}) => {
+  useEffect(() => {
+    Sentry.addBreadcrumb({
+      category: "navigation",
+      message: "CampaignInitialisationScreen"
+    });
+  }, []);
+
+  const { token, endpoint } = useAuthenticationContext();
+  const { fetchingState, updateCampaignConfig } = useUpdateCampaignConfig(
+    token,
+    endpoint
+  );
+  const checkVersion = useCheckVersion();
+  const checkUpdates = useCheckUpdates();
+
+  useEffect(() => {
+    updateCampaignConfig();
+    // updating the campaign config should only happen once when this screen is loaded
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const continueToNormalFlow = useCallback(() => {
+    const flowType = navigation.getParam("flowType", "DEFAULT");
+    switch (flowType) {
+      case "DEFAULT":
+        navigation.navigate("CollectCustomerDetailsScreen");
+        break;
+      case "MERCHANT":
+        navigation.navigate("MerchantPayoutScreen");
+        break;
+    }
+  }, [navigation]);
+
+  const [outdatedType, setOutdatedType] = useState<"BINARY" | "BUILD">();
+
+  const setState = useState()[1];
+  /**
+   * Tries up to RETRY_UPDATE_TIMES times to get the latest build version.
+   * If unable to retrieve it, don't allow the user to continue since the user
+   * has already failed the compatibility check.
+   */
+  const getNewBuildIfAny = useCallback(async (): Promise<void> => {
+    let lastUpdateResult;
+    for (let i = 0; i < RETRY_UPDATE_TIMES; i++) {
+      Sentry.addBreadcrumb({
+        category: "checkUpdates",
+        message: `attempt ${i + 1}`
+      });
+      const updateResult = await checkUpdates(true);
+      if (updateResult === "UPDATE_READY") {
```

Comment:
Would it be worth putting these constants in an object or enum? So we can refer to the same constant from both here and `useCheckUpdates`?

Reply:
Hmm this will throw a TypeScript error if "UPDATE_READY" is not within the union type "CheckUpdatesResult". So the union type restricts the values that `updateResult` can take—similar to a typical enum, but without the overhead of needing to do
```
enum CheckUpdatesResult {
  UPDATE_READY = "UPDATE_READY"
  ...
}
vs
type CheckUpdatesResult =
  | "TOO_RECENT_CHECK"
  | "NO_UPDATE"
  | "UPDATE_READY"
  | "UPDATE_ERROR";
```

Reply:
Didn't notice the type!! Nice nice